### PR TITLE
Automatically convert old presets to new format (but do not save it yet)

### PR DESCRIPTION
**Pull request by dimitry-**
Automatic conversion will be enabled after new format becomes human readable.

Bug: http://b/168799822
Test: Open old preset -> check preset folder, it now has .backup file
      and new preset file.

### CODE REVIEW

File: src/OrbitGl/App.cpp [hunk 1]
```diff
@@ -1619,23 +1620,35 @@ orbit_base::Future<ErrorMessageOr<void>> OrbitApp::LoadSymbols(
...
+  return module_data;
+}
+
+orbit_base::Future<ErrorMessageOr<void>> OrbitApp::LoadPresetModule(const std::string& module_path,
+                                                                    const PresetFile& preset_file) {
+  auto module_data_or_error = GetLoadedModuleByPath(module_path);
+
+  if (module_data_or_error.has_error()) {
+    return module_data_or_error.error();
+  }
```
Comment (pierricgimmig) [inline src/OrbitGl/App.cpp:1648]: OUTCOME_TRY?

File: src/OrbitGl/App.cpp [hunk 2]
```diff
@@ -2355,3 +2376,52 @@ void OrbitApp::SetTargetProcess(ProcessData* process) {
...
+
+  LOG("Converting preset file \"%s\" to new format.", preset_file.file_path().string());
+
+  // Convert first
+  PresetInfo new_info;
+  for (const auto& module_path : preset_file.GetModulePaths()) {
+    OUTCOME_TRY(module_data, GetLoadedModuleByPath(module_path.string()));
+    PresetModuleInfo module_info;
+
+    for (const auto function_hash :
```
Comment (pierricgimmig) [inline src/OrbitGl/App.cpp:2392]: auto -> uint64_t?

File: src/OrbitGl/App.cpp [hunk 3]
```diff
@@ -2355,3 +2376,52 @@ void OrbitApp::SetTargetProcess(ProcessData* process) {
...
+         preset_file.GetSelectedFunctionHashesForModuleLegacy(module_path)) {
+      const auto* function_info = module_data->FindFunctionFromHash(function_hash);
+      if (function_info == nullptr) {
+        ERROR("Could not find function hash %#x in module \"%s\"", function_hash, module_path);
+        continue;
+      }
+      module_info.add_function_names(function_info->pretty_name());
+    }
+
+    for (const auto function_hash :
```
Comment (pierricgimmig) [inline src/OrbitGl/App.cpp:2403]: auto -> uint64_t?

**Review by pierricgimmig — APPROVED** [review]
Cool, thanks for this Dimitry!

File: src/OrbitGl/App.cpp (hunk 1 above)
Comment (dimitry-) [reply src/OrbitGl/App.cpp:1648]: the method returns `Future<>` OUTCOME_TRY does not automatically convert it to the correct type.

File: src/OrbitGl/App.cpp (hunk 2 above)
Comment (dimitry-) [reply src/OrbitGl/App.cpp:2392]: done.

File: src/OrbitGl/App.cpp (hunk 3 above)
Comment (dimitry-) [reply src/OrbitGl/App.cpp:2403]: done.

File: src/OrbitGl/App.cpp (hunk 1 above)
Comment (pierricgimmig) [reply src/OrbitGl/App.cpp:1648]: ah interesting!

**Review by pierricgimmig — APPROVED** [review]

**Pull request merged**